web/src/server_events_dispatch.js

```diff
@@ -1022,6 +1023,14 @@ export function dispatch_normal_event(event) {
                 event,
                 message_events.update_current_view_for_topic_visibility(),
             );
+            // Discard cached message lists if `event` topic was / is followed.
+            if (
+                event.visibility_policy === user_topics.all_visibility_policies.FOLLOWED ||
+                user_topics.get_topic_visibility_policy(event.stream_id, event.topic_name) ===
+                    user_topics.all_visibility_policies.FOLLOWED
+            ) {
+                message_events.discard_cached_lists_with_term_type("is-followed");
```

Comment:
Does this second condition work? I think you need to do this before `user_topics_ui.handle_topic_updates`, or at least before `user_topics.set_user_topic(user_topic_event);` at the start of that function (which could reasonably be pulled up to sit here if helpful)

Reply:
This seems correct; please hand-test though:

``` diff
diff --git a/web/src/server_events_dispatch.js b/web/src/server_events_dispatch.js
index f94f50d0fa..7f99f99a1d 100644
--- a/web/src/server_events_dispatch.js
+++ b/web/src/server_events_dispatch.js
@@ -1018,7 +1018,11 @@ export function dispatch_normal_event(event) {
             }
             break;
 
-        case "user_topic":
+        case "user_topic": {
+            const previous_topic_visibility = user_topics.get_topic_visibility_policy(
+                event.stream_id,
+                event.topic_name,
+            );
             user_topics_ui.handle_topic_updates(
                 event,
                 message_events.update_current_view_for_topic_visibility(),
@@ -1026,11 +1030,11 @@ export function dispatch_normal_event(event) {
             // Discard cached message lists if `event` topic was / is followed.
             if (
                 event.visibility_policy === user_topics.all_visibility_policies.FOLLOWED ||
-                user_topics.get_topic_visibility_policy(event.stream_id, event.topic_name) ===
-                    user_topics.all_visibility_policies.FOLLOWED
+                previous_topic_visibility === user_topics.all_visibility_policies.FOLLOWED
             ) {
                 message_events.discard_cached_lists_with_term_type("is-followed");
             }
             break;
+        }
     }
 }
```

Reply:
oh yeah!